src/Listener/FileChecksumListener.php

```diff
@@ -0,0 +1,66 @@
+<?php
+/**
+ * This file is part of byrokrat\giroapp.
+ *
+ * byrokrat\giroapp is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * byrokrat\giroapp is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with byrokrat\giroapp. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright 2016-17 Hannes Forsgård
+ */
+
+declare(strict_types = 1);
+
+namespace byrokrat\giroapp\Listener;
+
+use hanneskod\yaysondb\Yaysondb;
+
+/**
+ * Filter NodeEvents where payee information does not match local registry
+ */
+class FileChecksumListener
+{
+    /**
+     * @var Yaysondb
+     */
+    private $db;
+
+    public function __construct(Yaysondb $db)
```

Comment:
Inject a `Mapper` instead. So one we decide to move to a different database backend we only need to update the mappers...